homeassistant/helpers/template/extensions/collection.py

```diff
@@ -170,6 +179,51 @@ def symmetric_difference(
 
         return list(set(value) ^ set(other))
 
+    def natural_sort(
+        self,
+        lst: Iterable[Any],
+        key: str | int | None = None,
+        alg: str | None = None,
+        reverse: bool = False,
+    ) -> list[Any]:
+        """Return elements sorted naturally."""
+
+        def _get_key(v: Any) -> Any:
+            """Get the key."""
+            if isinstance(key, str):
+                if isinstance(v, dict):
+                    return v.get(key)
+                return getattr(v, key, None)
+            if not isinstance(key, int):
+                raise KeyError(
+                    f"key expected an integer for list or string for list of lists, got {type(key).__name__}"
+                )
+            return v[key]
+
+        def _get_alg(expr: str | list[str]) -> int:
+            """Get the alg via an alg mask."""
+
+            ALG_MAP = {name: getattr(ns, name) for name in dir(ns) if name.isupper()}
```

Comment:
This can be lifted out of this scope to gain a little bit of performance.
We only need to calculate this once.